vignettes/rtestim.Rmd

```diff
@@ -0,0 +1,179 @@
+
+is the weighted sum of previous incidence at $t$, calculated by convolving the
+preceding $m$ days of new infections with the discretized serial interval distribution $w$ of length $m$. This delay distribution encapsulates the duration
+of time that a previous infection is likely to lead to future infection.
+
+To compute $\Rtseq$ with `{rtestim}`, the minimal information needed is the 
+new case counts at days up until $t$ and a parametric form for the serial 
+interval distribution (a Gamma density). By default, 
+$2.5$ is used for both the scale and shape parameters, based on the literature
+on contract tracing, representing the typical delay between case onsets.
+This is discretized to be supported on the integers. 
+The order of the difference operator, the degree of smoothness, defaults
+to $3$. 
+The sequence of smoothness penalty $\lambda$, if no $\lambda$ is provided, 
+is calculated internally by the algorithm.
+
+# Example - synthetic dataset
+
+## Quick start
+
+We first demonstrate the usage of the package on a synthetic dataset, where the
+new daily case counts are generated as a sequence of integers that starts from 1. 
+
+```{r}
+set.seed(12345)
+case_counts <- c(1, rpois(100, dnorm(1:100, 50, 15) * 500 + 1))
+plot(case_counts, xlab = "Time", ylab = "Case Counts")
+```
+
+We fit the model and visualize the resulting $\Rtseq$ with the following functions
+
+```{r}
+mod <- estimate_rt(observed_counts = case_counts, nsol = 20)
+plot(mod)
+```
+
+`{rtestim}` estimates a spectrum of $\Rtseq$s for a range of $\lambda$ values,
+where each $\Rtseq$ corresponds to a specific $\lambda$ value. If no $\lambda$ 
+value is supplied by the user, RtEstim will automatically calculate a sequence 
+of $\lambda$ values. The additional parameter `nsol = 20` specifies the number of 
+$\lambda$s for which the $\Rtseq$ is calculated
+
+
+## Cross Validation
+
+`{rtestim}` also provides a cross validation procedure for selecting the amount of smoothness to be used in the final estimate (leave-every-k-th-out cross validation). Minimizing this metric, in principle, balances prediction error
+and smoothness (`lambda.min`) though if smoother estimates are desired, one
+can instead use `lambda.1se`, the largest value of $\lambda$ within one standard
+error of the minimum.
+
+```{r}
+mod_cv <- cv_estimate_rt(observed_counts = case_counts, nsol = 20)
+```
+
+The following command plots the cross validation errors for each $\lambda$ in
+ascending order.
+
+```{r}
+plot(mod_cv)
+```
+
+The plot above displays vertical lines that correspond to the cross-validation scores 
+for specific values of $\lambda$. The blue point at the center of each line 
+represents the mean score for that value of $\lambda$ across all cross-validation
+folds. The top and bottom caps of each line indicate one cross-validation 
+standard error above and below the mean score for the given value of $\lambda$ 
+across all cross-validation folds. Two special values of $\lambda$'s are 
```

Comment:
I only see one dashed line for the largest lambda in the plot above.

Reply:
Unfortunately, this depends on the version of R. This time it should look better!

Reply:
Actually, I just realized that since we're not using linearized ADMM, we want to reverse the path. I've done that now and upped the number of iterations.